packages/tabs/src/styled/StyledTab.ts

```diff
@@ -5,55 +5,102 @@
  * found at http://www.apache.org/licenses/LICENSE-2.0.
  */
 
-import PropTypes from 'prop-types';
-import styled from 'styled-components';
-import classNames from 'classnames';
-import TabStyles from '@zendeskgarden/css-tabs';
-import { retrieveComponentStyles } from '@zendeskgarden/react-theming';
+import styled, { DefaultTheme, css, ThemeProps } from 'styled-components';
+import { retrieveComponentStyles, getColor, DEFAULT_THEME } from '@zendeskgarden/react-theming';
 
 const COMPONENT_ID = 'tabs.tab';
 
 interface IStyledTabProps {
-  isHovered?: boolean;
-  isFocused?: boolean;
-  isActive?: boolean;
   disabled?: boolean;
   isSelected?: boolean;
 }
 
+/**
+ * A high specificity is needed to apply the border-color in vertical orientations
+ */
+const colorStyles = ({ theme, isSelected }: IStyledTabProps & ThemeProps<DefaultTheme>) => css`
+  color: ${isSelected ? getColor('primaryHue', 600, theme) : 'inherit'};
+
+  &&& {
+    border-color: ${isSelected && 'currentColor'};
+  }
+
+  &:hover {
+    color: ${props => getColor('primaryHue', 600, props.theme)};
+  }
+
+  &:focus {
+    outline: none;
+    text-decoration: none;
+  }
+
+  &:active {
+    border-color: currentColor;
+    color: ${props => getColor('primaryHue', 600, props.theme)};
+  }
+
+  &::before {
+    transition: box-shadow 0.1s ease-in-out;
+    content: '';
+  }
+
+  &:active::before {
+    box-shadow: none;
+  }
+`;
+
+const focusStyles = ({ theme }: ThemeProps<DefaultTheme>) => css`
+  &[data-garden-focus-visible] {
+    color: ${getColor('primaryHue', 600, theme)};
```

Comment:
Since this component has to style the `::before` element I broke this out into a separate helper function. I found it easier to co-locate all of this special logic in a single helper.

Reply:
This helper has now been removed.